src/e2e-test/java/io/cdap/plugin/utils/PubSubClient.java

```diff
@@ -56,4 +104,186 @@ public static String getTopicCmekKey(String topicId) throws IOException {
+    IOException, ExecutionException, InterruptedException {
+    Encoding encoding = null;
+    TopicName topicName = TopicName.of(projectId, topicId);
+    // Get the topic encoding type.
+    try (TopicAdminClient topicAdminClient = TopicAdminClient.create()) {
+      encoding = topicAdminClient.getTopic(topicName).getSchemaSettings().getEncoding();
+    }
+
+    // Instantiate an avro-tools-generated class defined in `us-states.avsc`.
+    State state = State.newBuilder().setName("Alaska").setPostAbbr("AK").build();
+
+    Publisher publisher = null;
+
+    block:
+    try {
+      publisher = Publisher.newBuilder(topicName).build();
+
+      // Prepare to serialize the object to the output stream.
+      ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
+
+      Encoder encoder = null;
+
+      // Prepare an appropriate encoder for publishing to the topic.
+      switch (encoding) {
+        case BINARY:
+          System.out.println("Preparing a BINARY encoder...");
+          encoder = EncoderFactory.get().directBinaryEncoder(byteStream, /*reuse=*/ null);
+          break;
+
+        case JSON:
+          System.out.println("Preparing a JSON encoder...");
+          encoder = EncoderFactory.get().jsonEncoder(State.getClassSchema(), byteStream);
+          break;
+
+        default:
+          break block;
+      }
+
+      // Encode the object and write it to the output stream.
+      state.customEncode(encoder);
+      encoder.flush();
+
+      // Publish the encoded object as a Pub/Sub message.
+      ByteString data = ByteString.copyFrom(byteStream.toByteArray());
+      PubsubMessage message = PubsubMessage.newBuilder().setData(data).build();
+      System.out.println("Publishing message: " + message);
+
+      ApiFuture<String> future = publisher.publish(message);
+      System.out.println("Published message ID: " + future.get());
+
+    } finally {
+      if (publisher != null) {
+        publisher.shutdown();
+        publisher.awaitTermination(1, TimeUnit.MINUTES);
+      }
+    }
+  }
+
+  public static void deleteSchema(String projectId, String schemaId) throws IOException {
+    SchemaName schemaName = SchemaName.of(projectId, schemaId);
+
+    try (SchemaServiceClient schemaServiceClient = SchemaServiceClient.create()) {
+
+      schemaServiceClient.deleteSchema(schemaName);
+
+    } catch (NotFoundException e) {
+      System.out.println(schemaName + "not found.");
+    }
+  }
+
+  public static void deleteSubscription(String projectId, String subscriptionId) throws IOException {
+    try (SubscriptionAdminClient subscriptionAdminClient = SubscriptionAdminClient.create()) {
+      ProjectSubscriptionName subscriptionName = ProjectSubscriptionName.of(projectId, subscriptionId);
+      try {
+        subscriptionAdminClient.deleteSubscription(subscriptionName);
+      } catch (NotFoundException e) {
+        System.out.println(e.getMessage());
+      }
+    }
+  }
```

Comment:
Fix the code indentation and remove all the system.out.println

Reply:
done.